source/operations/checklists/hardware.rst

```diff
@@ -202,37 +202,54 @@ Storage
 
-      * - Type
-        - Cost
-        - Performance
-        - Tier
+   Format Drives as XFS
+   ++++++++++++++++++++
 
-      * - NVMe
-        - High
-        - High
-        - Hot
+   Format drives as XFS and present them to MinIO as a :abbr:`JBOD (Just a Bunch of Disks)` array with no RAID or other pooling configurations.
 
-      * - SSD
-        - Balanced
-        - Balanced
-        - Hot/Warm
+   MinIO **strongly recommends** disabling retry-on-error (``EIO``, ``ENOSPC``, and ``default``) behavior in XFS.
+   The default settings typically allow infinite retry-on-error and can produce unnecessary performance loss.
 
-      * - HDD
-        - Low
-        - Low
-        - Cold/Archival
+   The following script iterates through all drives at the specified mount path and sets recommended values for ``max_retries`` and ``retry_timeout_seconds``. 
+   Use this script on each node on which the MinIO Server runs.
+   Modify the ``/mnt/drive`` line to match the pattern used for your MinIO drives.
 
-   Format drives as XFS and present them to MinIO as a :abbr:`JBOD (Just a Bunch of Disks)` array with no RAID or other pooling configurations.
+   .. code-block:: bash
+      :class: copyable
+
+      #!/bin/bash
+
+      for i in $(df -h | grep /mnt/drive | awk '{ print $1 }'); do
+            mountPath="$(df -h | grep $i | awk '{ print $6 }')"
+            deviceName="$(basename $i)"
+            echo "Modifying xfs max_retries and retry_timeout_seconds for drive $i mounted at $mountPath"
+            echo 0 > /sys/fs/xfs/$deviceName/error/metadata/EIO/max_retries
+            echo 0 > /sys/fs/xfs/$deviceName/error/metadata/EIO/retry_timeout_seconds
+            echo 0 > /sys/fs/xfs/$deviceName/error/metadata/ENOSPC/max_retries
+            echo 0 > /sys/fs/xfs/$deviceName/error/metadata/ENOSPC/retry_timeout_seconds
+            echo 0 > /sys/fs/xfs/$deviceName/error/metadata/default/max_retries
+            echo 0 > /sys/fs/xfs/$deviceName/error/metadata/default/retry_timeout_seconds
+      done
+      exit 0
+
+   Since these settings do not persist on restart, you can use a ``cron`` job on ``@reboot`` to check each mounted drive at the designated path and set the necessary config.
+   Use ``crontab -e`` to create the following job, modifying the script path to match that on each node:
+
+   .. code-block:: shell
+      :class: copyable
+
+      @reboot /opt/minio/xfs-retry-settings.sh
+
+   Use Consistent Drive Type and Capacity
+   ++++++++++++++++++++++++++++++++++++++
 
-   Ensure a consistent drive type (NVMe, SSD, HDD) for the underyling storage. 
+   Ensure a consistent drive type (NVMe, SSD, HDD) for the underlying storage. 
    MinIO does not distinguish between storage types.
    Mixing storage types provides no benefit to MinIO.
```

Comment:
If there is "no benefit," what _is_ the implication of mixing types? We say the drives _should_ be the same type, but not what happens if they aren't. What are the consequences?

Reply:
@ravindk89 Is there a benefit to setting `retry_timeout_seconds`, if `max_retries` is already 0?